### ISSUE TITLE
datahub-upgrade fails on v0.8.14 

### ISSUE THREAD

**Issue by ybantes-esure** [acryldata/datahub-helm#39]
Hello, 

I am facing an issue while trying to upgrade datahub from v0.8.6 to 0.8.14 . When I run the datahub upgrade, I get the following error: 
`Executing Step 2/7: GMSQualificationStep...
java.io.FileNotFoundException: http://datahub-gms:8080/config
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1890)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	at com.linkedin.datahub.upgrade.nocode.GMSQualificationStep.lambda$executable$0(GMSQualificationStep.java:63)
	at com.linkedin.datahub.upgrade.impl.DefaultUpgradeManager.executeStepInternal(DefaultUpgradeManager.java:97)
	at com.linkedin.datahub.upgrade.impl.DefaultUpgradeManager.executeInternal(DefaultUpgradeManager.java:57)
	at com.linkedin.datahub.upgrade.impl.DefaultUpgradeManager.executeInternal(DefaultUpgradeManager.java:39)
	at com.linkedin.datahub.upgrade.impl.DefaultUpgradeManager.execute(DefaultUpgradeManager.java:30)
	at com.linkedin.datahub.upgrade.UpgradeCli.run(UpgradeCli.java:44)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:139)
	at com.linkedin.datahub.upgrade.UpgradeCliApplication.main(UpgradeCliApplication.java:13)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
ERROR: Cannot connect to GMSat host datahub-gms port 8080. Make sure GMS is on the latest version and is running at that host before starting the migration.`

The service is there and it listens on port 8080.  I am also getting 404 when I try to curl this endpoint: http://datahub-gms:8080/config 

But the service is there. 

-------------------------------------
If I run the upgrade job on the old version of GMS(0.8.6) I am getting the following error: 
` :: Spring Boot ::        (v2.1.4.RELEASE)
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
SLF4J: Failed to load class "org.slf4j.impl.StaticMDCBinder".
SLF4J: Defaulting to no-operation MDCAdapter implementation.
SLF4J: See http://www.slf4j.org/codes.html#no_static_mdc_binder for further details.
Sep 29, 2021 7:28:17 AM org.neo4j.driver.internal.logging.JULogger info
INFO: Direct driver instance 583015088 created for server address localhost:7687
Starting upgrade with id NoCodeDataMigration...
Executing Step 1/7: RemoveAspectV2TableStep...
Completed Step 1/7: RemoveAspectV2TableStep successfully.
Executing Step 2/7: GMSQualificationStep...
Completed Step 2/7: GMSQualificationStep successfully.
Executing Step 3/7: UpgradeQualificationStep...
-- V1 table does not exist
Failed to qualify upgrade candidate. Aborting the upgrade...
Step with id UpgradeQualificationStep requested an abort of the in-progress update. Aborting the upgrade...
Upgrade NoCodeDataMigration completed with result ABORTED. Exiting...`

I am not really sure if it is a bug or if I am missing anything. Any ideas here?

**Comment by phildum** [comment]
I had the same issue with a fresh install of datahub, so this time I could just disable the upgrade job in values.yaml.

I got your first issue, because the upgrade job started before the GMS pod had come up. A few pods failed, but then the GMS came up and that is when I encountered your second issue. This seems to be because the upgrade job could not find the original `metadata_aspect` table. https://github.com/linkedin/datahub/blob/master/datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/UpgradeQualificationStep.java#L57 -> https://github.com/linkedin/datahub/blob/master/metadata-io/src/main/java/com/linkedin/metadata/entity/AspectStorageValidationUtil.java#L44

In my case, this makes sense, because it is a fresh install.

However, it does not seem possible to specify which upgrade jobs should be run in values.yaml. It seems to be all or nothing?
